spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/autoconfig/GcpProperties.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2017 original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.core.autoconfig;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * @author João André Martins
+ */
+@ConfigurationProperties("cloud.gcp")
+public class GcpProperties {
+
+	private String projectId;
```

Comment:
Ideally, we'd have cloud.gcp.project.id instead of cloud.gcp.projectId. Not sure if there's a way to make this happen. Maybe nested configurations or a Project/Credentials object?